plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -169,6 +169,23 @@ func findNodesThatFit(pod *api.Pod, nodeNameToInfo map[string]*schedulercache.No
 			if err != nil {
 				return api.NodeList{}, FailedPredicateMap{}, err
 			}
+			//add the extender failed info to failedPredicateMap
+			for _, filteredNode := range filtered {
```

Comment:
Extender.Filter return error with error information, while failed because node is not fit (non-compliant), returning fitNodeList is enough. We can get the failed nodes easily, needn't all the extenders to achieve returning the failed nodes, I think so.